File: task-03/Haskell/Sub-Task 3/subtask3.hs
import Control.Monad (forM_)
printDiamond :: Int -> IO ()
printDiamond n = do
    forM_ [0..n-1] $ \i -> putStrLn $ replicate (n - i - 1) ' ' ++ replicate (2 * i + 1) '*'
    forM_ [n-2,n-3..0] $ \i -> putStrLn $ replicate (n - i - 1) ' ' ++ replicate (2 * i + 1) '*'

main = do
    putStrLn "Enter a number: "
    n <- readLn
    printDiamond n

File: task-03/C++/Sub-Task 3/subtask3.cpp
#include <iostream>
#include <string>
void printDiamond(int n) {
    for (int i = 0; i < n; i++) {
        std::cout << std::string(n - i - 1, ' ') << std::string(2 * i + 1, '*') << std::endl;
    }
    for (int i = n - 2; i >= 0; i--) {
        std::cout << std::string(n - i - 1, ' ') << std::string(2 * i + 1, '*') << std::endl;
    }
}
int main() {
    int n;
    std::cout << "Enter a number: ";
    std::cin >> n;
    printDiamond(n);
    return 0;
}